File: prs-ui/apps/sadarr/src/app/authentication/models/user.model.ts
export interface UserModel {
  id: string;
  firstName: string;
  lastName: string;
  email: string;
  emailConfirmed: boolean;
  token: string;
  userRoles: UserRoleModel[];
}

export interface UserRoleModel {
  userId: string;
  roleId: string;
  role: RoleModel;
}

export interface RoleModel {
  id: string;
  name: string;
}

File: prs-api/Models/Dtos/UserModel.cs
using prs_api.Data;

namespace prs_api.Models.Dtos
{
    public class UserModel
    {
        public Guid Id { get; set; }
        public string Email { get; set; } = string.Empty;
        public bool IsEmailConfirmed { get; set; }
        public string FirstName { get; set; } = string.Empty;
        public string LastName { get; set; } = string.Empty;
        public List<UserRoleModel> UserRoles { get; set; } = new List<UserRoleModel>();
        public List<MovieRequestModel> MovieRequests { get; set; } = new List<MovieRequestModel>();
    }

    public class UserRoleModel
    {
        public Guid UserId { get; set; }
        public Guid RoleId { get; set; }

        public UserModel? User { get; set; } = null;
        public RoleModel? Role { get; set; } = null;
    }

    public class RoleModel 
    {
        public Guid Id { get; set; }
        public string Name { get; set; } = string.Empty;

        public List<UserRole> UserRoles { get; set; } = new List<UserRole>();
    }
}
